play-json/src/main/scala/com/lightbend/lagom/scaladsl/playjson/JsonSerializerRegistry.scala

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.scaladsl.playjson
+
+import akka.actor.ExtendedActorSystem
+import akka.actor.setup.ActorSystemSetup
+import akka.serialization.{ SerializationSetup, SerializerDetails }
+
+import scala.collection.immutable
+
+/**
+ * Create a concrete subclass of this and register using the `lagom.serialization.play-json.serializer-registry`
+ * setting to have your play-json serializers picked up and used by Lagom.
+ */
+abstract class JsonSerializerRegistry {
+
+  def serializers: immutable.Seq[JsonSerializer[_]]
+
+  /**
+   * A set of migrations keyed by the fully classified class name that the migration should be triggered for
+   */
+  def migrations: Map[String, JsonMigration] = Map.empty
+
+  /**
+   * Concatenate the serializers and migrations of this registry with another registry to form a new registry.
+   */
+  final def ++(other: JsonSerializerRegistry): JsonSerializerRegistry = {
+    val self = this
+    new JsonSerializerRegistry {
+      override def serializers = self.serializers ++ other.serializers
+      override def migrations = self.migrations ++ other.migrations
+    }
+  }
+}
+
+object JsonSerializerRegistry {
+  /**
+   * Create the serializer details for the given serializer registry.
+   */
+  def serializerDetailsFor(system: ExtendedActorSystem, registry: JsonSerializerRegistry): SerializerDetails = {
+    SerializerDetails(
+      "lagom-play-json",
+      new PlayJsonSerializer(system, registry),
+      registry.serializers.map(_.entityClass)
+    )
+  }
+
+  /**
+   * Create a serializer setup for the given serializer registry.
+   *
+   * This is only useful if you only want to register the play-json serializer and its registry, if you want to
+   * register other Akka serializers, then you should create your own serialization setup.
+   */
+  def serializationSetupFor(registry: JsonSerializerRegistry): SerializationSetup = {
+    SerializationSetup { system =>
+      Vector(serializerDetailsFor(system, registry))
+    }
+  }
+
+  /**
+   * Create an actor system setup for the given serializer registry.
+   *
+   * This is only useful if you don't want to modify the configuration of the actor system in any way, if you do,
+   * then you should instead use [[#serializationSetupFor]], and manually create your own [[ActorSystemSetup]] from
+   * that.
+   */
+  def actorSystemSetupFor(registry: JsonSerializerRegistry): ActorSystemSetup = {
+    ActorSystemSetup(serializationSetupFor(registry))
+  }
```

Comment:
Convenience methods here, especially for end users to use in tests for example when they have to create their own actor system.